cassovary-core/src/main/scala/com/twitter/cassovary/graph/DirectedPathCollection.scala

```diff
@@ -148,10 +147,18 @@ class PathCounterComparator(pathCountsPerId: Int2ObjectOpenHashMap[Object2IntOpe
   override def compare(dp1: DirectedPath, dp2: DirectedPath): Int = {
     val dp1Count = infoMap.getInt(dp1)
     val dp2Count = infoMap.getInt(dp2)
-    if (descending) {
-      dp2Count - dp1Count
+    if (dp1Count != dp2Count) {
+      if (descending) {
+        dp2Count - dp1Count
+      } else {
+        dp1Count - dp2Count
+      }
     } else {
-      dp1Count - dp2Count
+      if (descending) {
+        dp2.length - dp1.length
```

Comment:
might be counter-intuitive, but shorter paths are generally better, so I would reverse this -- i.e., put dp1.length - dp2.length here and put a brief comment here as well